client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -278,10 +373,22 @@ public void performRequestAsync(String method, String endpoint, Map<String, Stri
      * @param entity the body of the request, null if not applicable
      * @param responseListener the {@link ResponseListener} to notify when the request is completed or fails
      * @param headers the optional request headers
+     * @deprecated Prefer {@link #performRequestAsync(Request, ResponseListener)}
      */
+    @Deprecated
     public void performRequestAsync(String method, String endpoint, Map<String, String> params,
                                     HttpEntity entity, ResponseListener responseListener, Header... headers) {
-        performRequestAsync(method, endpoint, params, entity, HttpAsyncResponseConsumerFactory.DEFAULT, responseListener, headers);
+        Request request;
+        try {
+            request = new Request(method, endpoint);
+            request.setParameters(params);
+            request.setEntity(entity);
+            request.setHeaders(headers);
+        } catch (Exception e) {
+            responseListener.onFailure(e);
+            return;
+        }
```

Comment:
shall we have a small private method that allows to build the request given all the different params and incorporates the catch as well?

Reply:
I didn't do that because each method sets different stuff.